src/py42/services/alerts.py

```diff
@@ -12,16 +12,35 @@ class AlertService(BaseService):
 
     _CREATED_AT = u"CreatedAt"
     _RULE_METADATA = u"ruleMetadata"
+    _SEARCH_KEY = u"alerts"
 
     def __init__(self, connection, user_context):
         super(AlertService, self).__init__(connection)
         self._user_context = user_context
 
-    def search(self, query):
+    def search(self, query, page_num=1, page_size=None):
+        query.page_number = page_num - 1
```

Comment:
What if the `page_num` and `page_size` are already set on the query?

Reply:
By default they are set in query, so if user has passed on `page_num` and `page_size` as args, it will overwrite those defaults.

Reply:
As per existing implementation, the user need not set them in query, they get set internally(its an implementation detail) with default value, unknown to user. So now we provide an option by which they pass it as separate arguments and how we set them internally isn't exposed to them.